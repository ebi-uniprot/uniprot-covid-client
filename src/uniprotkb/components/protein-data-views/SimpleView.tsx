import React from 'react';
import { Link } from 'react-router-dom';

const SimpleView = (props: { termValue: string; linkTo?: string }) => {
  const { termValue, linkTo } = props;
  if (typeof linkTo !== 'undefined') {
    return <Link to={linkTo}>{termValue}</Link>;
  } 
    return <span>{termValue}</span>;
  
};

export default SimpleView;
