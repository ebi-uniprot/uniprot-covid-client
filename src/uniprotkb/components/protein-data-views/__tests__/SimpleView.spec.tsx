import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import SimpleView from '../SimpleView';

describe('SimpleView component', () => {
  test('should render', () => {
    const { asFragment } = render(<SimpleView termValue="blah" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render with link', () => {
    const { asFragment } = render(
      <Router>
        <SimpleView termValue="blah" linkTo="linkto" />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
