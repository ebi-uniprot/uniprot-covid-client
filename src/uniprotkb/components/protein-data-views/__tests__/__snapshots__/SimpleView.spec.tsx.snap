// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleView component should render 1`] = `
<DocumentFragment>
  <span>
    blah
  </span>
</DocumentFragment>
`;

exports[`SimpleView component should render with link 1`] = `
<DocumentFragment>
  <a
    href="/linkto"
  >
    blah
  </a>
</DocumentFragment>
`;
